beetsplug/smartplaylist.py

```diff
@@ -15,76 +15,22 @@
 """Generates smart playlists based on beets queries.
 """
 from __future__ import print_function
+from itertools import chain
 
 from beets.plugins import BeetsPlugin
-from beets import config, ui, library
+from beets import ui
 from beets.util import normpath, syspath
 import os
 
-# Global variable so that smartplaylist can detect database changes and run
-# only once before beets exits.
-database_changed = False
 
-
-def _items_for_query(lib, playlist, album=False):
-    """Get the matching items for a playlist's configured queries.
-    `album` indicates whether to process the item-level query or the
-    album-level query (if any).
-    """
-    key = 'album_query' if album else 'query'
-    if key not in playlist:
-        return []
-
-    # Parse quer(ies). If it's a list, perform the queries and manually
-    # concatenate the results
-    query_strings = playlist[key]
-    if not isinstance(query_strings, (list, tuple)):
-        query_strings = [query_strings]
-    model = library.Album if album else library.Item
-    results = []
-    for q in query_strings:
-        query, sort = library.parse_query_string(q, model)
-        if album:
-            new = lib.albums(query, sort)
-        else:
-            new = lib.items(query, sort)
-        results.extend(new)
-    return results
-
-
-def update_playlists(lib):
-    ui.print_("Updating smart playlists...")
-    playlists = config['smartplaylist']['playlists'].get(list)
-    playlist_dir = config['smartplaylist']['playlist_dir'].as_filename()
-    relative_to = config['smartplaylist']['relative_to'].get()
-    if relative_to:
-        relative_to = normpath(relative_to)
-
-    for playlist in playlists:
-        items = []
-        items.extend(_items_for_query(lib, playlist, True))
-        items.extend(_items_for_query(lib, playlist, False))
-
-        m3us = {}
-        basename = playlist['name'].encode('utf8')
-        # As we allow tags in the m3u names, we'll need to iterate through
-        # the items and generate the correct m3u file names.
-        for item in items:
-            m3u_name = item.evaluate_template(basename, True)
-            if not (m3u_name in m3us):
-                m3us[m3u_name] = []
-            item_path = item.path
-            if relative_to:
-                item_path = os.path.relpath(item.path, relative_to)
-            if item_path not in m3us[m3u_name]:
-                m3us[m3u_name].append(item_path)
-        # Now iterate through the m3us that we need to generate
-        for m3u in m3us:
-            m3u_path = normpath(os.path.join(playlist_dir, m3u))
-            with open(syspath(m3u_path), 'w') as f:
-                for path in m3us[m3u]:
-                    f.write(path + '\n')
-    ui.print_("... Done")
+def _items_for_query(lib, queries, album):
+    """Get the matching items for a query.
+    `album` indicates whether the queries are item-level or album-level"""
```

Comment:
Tiny doc string style nitpick: end the sentence with a period . and put the """ on its own line.